code/modules/mob/living/living.dm

```diff
@@ -837,30 +837,35 @@ Thanks.
 			L.visible_message("<span class='danger'>[L] resists!</span>")
 
 
-	//unbuckling yourself
-	if(L.locked_to && L.special_delayer.blocked() && istype(L.locked_to, /obj/structure/bed))
-		var/obj/structure/bed/B = L.locked_to
-		if(iscarbon(L))
-			var/mob/living/carbon/C = L
-			if(C.handcuffed)
-				C.delayNextAttack(100)
-				C.delayNextSpecial(100)
-				C.visible_message("<span class='warning'>[C] attempts to unbuckle themself!</span>",
-								  "<span class='warning'>You attempt to unbuckle yourself (this will take around two minutes, and you need to stay still).</span>",
-								  self_drugged_message="<span class='warning'>You attempt to regain control of your legs (this will take a while).</span>")
-				spawn(0)
-					if(do_after(usr, usr, 1200))
-						if(!C.locked_to)
-							return
-						C.visible_message("<span class='danger'>[C] manages to unbuckle themself!</span>",\
-							"<span class='notice'>You successfully unbuckle yourself.</span>",\
-							self_drugged_message="<span class='notice'>You successfully regain control of your legs and stand up.</span>")
-						B.manual_unbuckle(C)
-					else
-						C.simple_message("<span class='warning'>Your unbuckling attempt was interrupted.</span>", \
-							"<span class='warning'>Your attempt to regain control of your legs was interrupted. Damn it!</span>")
-		else
-			B.manual_unbuckle(L)
+	if(L.locked_to && L.special_delayer.blocked())
+		//unbuckling yourself
+		if(istype(L.locked_to, /obj/structure/bed))
+			var/obj/structure/bed/B = L.locked_to
+			if(iscarbon(L))
+				var/mob/living/carbon/C = L
+				if(C.handcuffed)
+					C.delayNextAttack(100)
+					C.delayNextSpecial(100)
+					C.visible_message("<span class='warning'>[C] attempts to unbuckle themself!</span>",
+									  "<span class='warning'>You attempt to unbuckle yourself (this will take around two minutes, and you need to stay still).</span>",
+									  self_drugged_message="<span class='warning'>You attempt to regain control of your legs (this will take a while).</span>")
+					spawn(0)
+						if(do_after(usr, usr, 1200))
+							if(!C.locked_to)
+								return
+							C.visible_message("<span class='danger'>[C] manages to unbuckle themself!</span>",\
+								"<span class='notice'>You successfully unbuckle yourself.</span>",\
+								self_drugged_message="<span class='notice'>You successfully regain control of your legs and stand up.</span>")
+							B.manual_unbuckle(C)
+						else
+							C.simple_message("<span class='warning'>Your unbuckling attempt was interrupted.</span>", \
+								"<span class='warning'>Your attempt to regain control of your legs was interrupted. Damn it!</span>")
+			else
+				B.manual_unbuckle(L)
+		//release from kudzu
+		else if(istype(L.locked_to, /obj/effect/plantsegment))
```

Comment:
Aaaaaaah, terrible snowflaking.

Alright, there may not appear to be a very obvious solution to this, but there is:

First you'll have to define an event hook for mobs, `on_resist` or something like that:

Put this somewhere:

``` DM
/mob/living
	var/event/on_resist

/mob/living/New()
	. = ..()
	on_resist = new(owner = src)

/mob/living/Destroy()
	. = ..()
	qdel(on_resist)
	on_resist = null
```

Now, somewhere, probably at the beginning of this proc, put the following code:

``` DM
INVOKE_EVENT(on_resist, list())
```

You'll need to define a new variable on the plants to store the event key (this can be ignored but it's not all that clean, if you do you'll need to rebuild the key manually, it's `"\ref[objectRef]:[procName]"`):

``` DM
/obj/effect/plantsegment
	var/tmp/on_resist_key
```

Now, add the following to the code that happens when the mob is **captured** by the plants:

``` DM
on_resist_key = mob.on_resist.Add(src, "manual_unbuckle")
```

And the code for when the mob gets released (Note: just to be sure, I recommend making this part of `unlock_atom()`):

``` DM
mob.on_resist.Remove(on_resist_key)
on_resist_key = null
```

If all goes well, this'll cause the plant to be notified when the mob is resisting, in a non shit fashion!

Note: this won't pass down a user to `manual_unbuckle()`, so you might want to keep that in mind.

Well done, you now have the highest quality mob code in the codebase.